mypy/subtypes.py

```diff
@@ -84,6 +85,12 @@
 IS_VAR: Final = 4
 IS_EXPLICIT_SETTER: Final = 5
 
+# Known unsafe subtyping relationships that should trigger warnings.
+# Each tuple is (subclass_fullname, superclass_fullname).
+# These are cases where a class is a subclass at runtime but treating it
+# as a subtype can cause runtime errors.
+UNSAFE_SUBTYPING_PAIRS: Final = [("datetime.datetime", "datetime.date")]
```

Comment:
This could be a set, but I'm not sure whether it is actually faster than a single-element list